File: Embarcadero/Berlin/CPP/SearchContext/dataData.cpp
//---------------------------------------------------------------------------

// This software is Copyright (c) 2015 Embarcadero Technologies, Inc.
// You may only use this software if you are an authorized licensee
// of an Embarcadero developer tools product.
// This software is considered a Redistributable as defined under
// the software license agreement that comes with the Embarcadero Products
// and is subject to that software license agreement.

//---------------------------------------------------------------------------

#pragma hdrstop

#include "dataData.h"
#include "FMX.Dialogs.hpp"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma classgroup "System.Classes.TPersistent"
#pragma resource "*.dfm"
TdtmdlData *dtmdlData;
//---------------------------------------------------------------------------
String FileName()
{
   return(ExtractFilePath(GetModuleName(0))+"iconData.cds");
}
//---------------------------------------------------------------------------
__fastcall TdtmdlData::TdtmdlData(TComponent* Owner)
	: TDataModule(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TdtmdlData::DataModuleCreate(TObject *Sender)
{
	this->cdsIconData->FileName = FileName();

	if (FileExists(this->cdsIconData->FileName)) {
		try {
			cdsIconData->Active = True;
		} catch (...) {
			cdsIconData->Close();
			cdsIconData->CreateDataSet();
		}
	}
	else
	{
		cdsIconData->CreateDataSet();
	}

}

//---------------------------------------------------------------------------
void __fastcall TdtmdlData::cdsIconDataAfterPost(TDataSet *DataSet)
{
	cdsIconData->SaveToFile();
}
//---------------------------------------------------------------------------

File: Embarcadero/Rio/Object Pascal/SearchContext/dataData.pas
//---------------------------------------------------------------------------

// This software is Copyright (c) 2015 Embarcadero Technologies, Inc.
// You may only use this software if you are an authorized licensee
// of an Embarcadero developer tools product.
// This software is considered a Redistributable as defined under
// the software license agreement that comes with the Embarcadero Products
// and is subject to that software license agreement.

//---------------------------------------------------------------------------

unit dataData;

interface

uses
  {$IFDEF Windows}
   MidasLib,
  {$ENDIF}
  System.SysUtils, System.Classes, Data.DB, Datasnap.DBClient;

type
  TdtmdlData = class(TDataModule)
    cdsIconData: TClientDataSet;
    dsIconData: TDataSource;
    cdsIconDataID: TIntegerField;
    cdsIconDataCategory: TStringField;
    cdsIconDataDescription: TStringField;
    cdsIconDataSearchTerms: TMemoField;
    cdsIconDataIcon: TBlobField;
    procedure DataModuleCreate(Sender: TObject);
    procedure cdsIconDataAfterPost(DataSet: TDataSet);
  private
    { Private declarations }
    function FileName : string;
  public
    { Public declarations }
  end;

var
  dtmdlData: TdtmdlData;

implementation

{%CLASSGROUP 'System.Classes.TPersistent'}

{$R *.dfm}

procedure TdtmdlData.cdsIconDataAfterPost(DataSet: TDataSet);
begin
  cdsIconData.SaveToFile();
end;

procedure TdtmdlData.DataModuleCreate(Sender: TObject);
begin
  cdsIconData.FileName := FileName;

  if not FileExists(FileName) then
    cdsIconData.CreateDataSet
  else
    cdsIconData.Active := True;
end;

function TdtmdlData.FileName: string;
begin
 Result := ExtractFilePath(ParamStr(0))+'iconData.cds';
end;

end.
